File: Task/Sort-the-letters-of-string-in-alphabetical-order/Delphi/sort-the-letters-of-string-in-alphabetical-order.pas
function SortStringLetters(S: string): string;
{Extract letters from string and sort them}
{Uses string list component to hold and sort chars}
var SL: TStringList;
var I: integer;
begin
SL:=TStringList.Create;
try
for I:=1 to Length(S) do SL.Add(S[I]);
SL.Sort;
SetLength(Result,SL.Count);
for I:=0 to SL.Count-1 do
 Result[I+1]:=SL[I][1];
finally SL.Free; end;
end;

const S1 = 'Now is the time for all good men to come to the aid of the party.';
const S2 = 'See the quick brown fox jump over the lazy dog.';

procedure ShowSortedLetters(Memo: TMemo);
var S: string;
begin
Memo.Lines.Add('Unsorted = '+S1);
Memo.Lines.Add('Sorted = '+SortStringLetters(S1));
Memo.Lines.Add('Unsorted = '+S2);
Memo.Lines.Add('Sorted = '+SortStringLetters(S2));
end;

File: Task/Sort-the-letters-of-string-in-alphabetical-order/Java/sort-the-letters-of-string-in-alphabetical-order.java
import java.util.List;
import java.util.function.BiConsumer;
import java.util.function.Function;
import java.util.stream.IntStream;

public final class SortTheLettersOfStringInAlphabeticalOrder {

	public static void main(String[] args) {		
		Function<String, String> bubbleSort = ( text -> {
			char[] chars = text.toCharArray();
		    BiConsumer<char[], Integer> swapValueIf = (array, i) -> {
		        if ( array[i] > array[i + 1] ) {
		            final char temp = array[i];
		            chars[i] = array[i + 1];
		            chars[i + 1] = temp;
		        }
		    };

		    IntStream.range(0, chars.length - 1)
		    	.forEach( i -> IntStream.range(0, chars.length - 1)
		            .forEach( j -> swapValueIf.accept(chars, j) ) );
		
		    return new String(chars);
		} );
		
		List<String> tests = List.of( "Now is the time for all good men to come to the aid of the party.",
				                      "The quick brown dog jumped over the lazy fox." );
		
		tests.forEach( test -> {
			System.out.println(test + " =>");
			System.out.println(bubbleSort.apply(test));
		} );
	}
	
}
